src/controllers/auth.controller.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+
+const { User } = require('../models/user.js');
+const { ApiError } = require('../exeptions/api.error.js');
+const userService = require('../services/user.service.js');
+const jwtService = require('../services/jwt.service.js');
+const tokenService = require('../services/token.service.js');
+const emailService = require('../services/email.service.js');
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async(req, res) => {
```

Comment:
please add try/catch blocks to all controller handlers for this controller and user.controller and return `400` error code on catch